homeassistant/components/vacuum/__init__.py

```diff
@@ -127,24 +127,71 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     await component.async_setup(config)
 
-    component.async_register_entity_service(SERVICE_TURN_ON, {}, "async_turn_on")
-    component.async_register_entity_service(SERVICE_TURN_OFF, {}, "async_turn_off")
-    component.async_register_entity_service(SERVICE_TOGGLE, {}, "async_toggle")
     component.async_register_entity_service(
-        SERVICE_START_PAUSE, {}, "async_start_pause"
+        SERVICE_TURN_ON,
+        {},
+        "async_turn_on",
+        [VacuumEntityFeature.TURN_ON],
     )
-    component.async_register_entity_service(SERVICE_START, {}, "async_start")
-    component.async_register_entity_service(SERVICE_PAUSE, {}, "async_pause")
     component.async_register_entity_service(
-        SERVICE_RETURN_TO_BASE, {}, "async_return_to_base"
+        SERVICE_TURN_OFF,
+        {},
+        "async_turn_off",
+        [VacuumEntityFeature.TURN_OFF],
+    )
+    component.async_register_entity_service(
+        SERVICE_TOGGLE,
+        {},
+        "async_toggle",
+        [VacuumEntityFeature.TURN_OFF | VacuumEntityFeature.TURN_ON],
+    )
+    component.async_register_entity_service(
+        SERVICE_START_PAUSE,
+        {},
+        "async_start_pause",
+        [VacuumEntityFeature.PAUSE],
```

Comment:
We need both start and pause here, right?

Reply:
No, VacuumEntityFeature.START signals support for the `vacuum.start` service supported only by `StateVacuumEntity`, see https://github.com/home-assistant/core/pull/15751

Reply:
Ok. Can we add a comment above this registration that this is a legacy service which only requires pause as supported feature?